### PR TITLE
Add event cancellation

### PR DESCRIPTION
**Pull request by CJ4n**

### CODE REVIEW

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 1]
```diff
@@ -54,27 +54,34 @@ public ResponseEntity<Event> addEvent(@ApiParam(value = "Add event") @Valid @Req
   }
 
   /**
-   *
    * User needs to be authorized
-   *
+   * <p>
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: ?

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 2]
```diff
@@ -83,13 +90,11 @@ public ResponseEntity<List<Event>> getByCategory(
   }
 
   /**
-   * GET /events/{id} : Find event by ID
-   * Returns a single event
+   * GET /events/{id} : Find event by ID Returns a single event
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: Why have you changed that ?

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 3]
```diff
@@ -83,13 +90,11 @@ public ResponseEntity<List<Event>> getByCategory(
...
    *
    * @param id ID of event to return (required)
-   * @return successful operation (status code 200)
-   *         or Invalid ID supplied (status code 400)
-   *         or Event not found (status code 404)
+   * @return successful operation (status code 200) or Invalid ID supplied (status code 400) or
+   * Event not found (status code 404)
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: Why have you changed that ?

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 4]
```diff
@@ -54,27 +54,34 @@ public ResponseEntity<Event> addEvent(@ApiParam(value = "Add event") @Valid @Req
...
 
   /**
    * GET /events/getByCategory : Return list of all events in category
    *
    * @param categoryId ID of category (required)
-   * @return successful operation (status code 200)
-   *         or Invalid category ID supplied (status code 400)
+   * @return successful operation (status code 200) or Invalid category ID supplied (status code
+   * 400)
    */
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: Why have you changed that ?

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 5]
```diff
@@ -54,27 +54,34 @@ public ResponseEntity<Event> addEvent(@ApiParam(value = "Add event") @Valid @Req
...
-   *
    * User needs to be authorized
-   *
+   * <p>
    * DELETE /events/{id} : Cancel event
    *
    * @param id id of Event (required)
-   * @return deleted (status code 204)
-   *         or id not found (status code 404)
+   * @return deleted (status code 204) or id not found (status code 404)
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: Why have you changed that ?

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 6]
```diff
@@ -110,21 +115,22 @@ public ResponseEntity<EventWithPlaces> getEventById(
...
   @Override
   public ResponseEntity<List<Event>> getEvents() {
-    return new ResponseEntity<>(eventService.getAllEvents(),HttpStatus.OK);
+    return new ResponseEntity<>(eventService.getAllEvents(), HttpStatus.OK);
   }
 
   /**
-   *
    * User needs to be authorized
-   *
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: ?

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 7]
```diff
@@ -110,21 +115,22 @@ public ResponseEntity<EventWithPlaces> getEventById(
...
+   * <p>
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: ?

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java [hunk 8]
```diff
@@ -110,21 +115,22 @@ public ResponseEntity<EventWithPlaces> getEventById(
...
    *
    * @return successful operation (status code 200)
    */
   @Override
   public ResponseEntity<List<Event>> getMyEvents() {
     UserDetails userDetails = getUserDetails();
-    if(userDetails == null) return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    if (userDetails == null) {
+      return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/EventsApiImpl.java]: Why have you changed that ?

File: src/main/java/com/team3/central/services/EventService.java [hunk 9]
```diff
@@ -81,4 +83,21 @@ public List<com.team3.central.openapi.model.Event> getForUser(String email) {
         .map(eventMapper::convertToModel)
         .collect(Collectors.toList());
   }
+
+  // TODO: Do we notify the user about cancellation?
```
Comment (adamdab) [inline src/main/java/com/team3/central/services/EventService.java]: We don't, could you remove this TODO

File: src/test/java/com/team3/central/services/EventServiceTest.java [hunk 10]
```diff
@@ -247,4 +263,98 @@ void getForUser() {
     // then
     assertThat(res).isNotEmpty();
   }
+
+  @Test
+  public void deleteEvent_shouldReturnFalse_whenEventNotFound() {
+    // given
```
Comment (adamdab) [inline src/test/java/com/team3/central/services/EventServiceTest.java]: We need to make naming convention consistent, change this naming style here or for all tests please

File: src/test/java/com/team3/central/services/EventServiceTest.java [hunk 11]
```diff
@@ -247,4 +263,98 @@ void getForUser() {
...
+    // when
+    boolean result = eventService.deleteEvent(EVENT_ID, ORGANIZER_EMAIL);
+
+    // then
+    assertFalse(result);
+  }
+
+  @Test
+  public void deleteEvent_shouldReturnFalse_whenOrganizerEmailMismatch() {
+    // given
```
Comment (adamdab) [inline src/test/java/com/team3/central/services/EventServiceTest.java]: We need to make naming convention consistent, change this naming style here or for all tests please

File: src/test/java/com/team3/central/services/EventServiceTest.java [hunk 12]
```diff
@@ -247,4 +263,98 @@ void getForUser() {
...
+    boolean result = eventService.deleteEvent(EVENT_ID, ORGANIZER_EMAIL);
+
+    // then
+    assertFalse(result);
+  }
+
+  @ParameterizedTest
+  @MethodSource("testData")
+  public void deleteEvent_shouldReturnFalse_whenEventAlreadyCancelled(EventStatus status) {
+    // given
```
Comment (adamdab) [inline src/test/java/com/team3/central/services/EventServiceTest.java]: We need to make naming convention consistent, change this naming style here or for all tests please

File: src/test/java/com/team3/central/services/EventServiceTest.java [hunk 13]
```diff
@@ -247,4 +263,98 @@ void getForUser() {
...
+    // when
+    boolean result = eventService.deleteEvent(EVENT_ID, ORGANIZER_EMAIL);
+
+    // then
+    assertFalse(result);
+  }
+
+  @Test
+  public void deleteEvent_shouldCancelEvent_whenValidRequest() {
+    // given
```
Comment (adamdab) [inline src/test/java/com/team3/central/services/EventServiceTest.java]: We need to make naming convention consistent, change this naming style here or for all tests please

File: src/test/java/com/team3/central/services/EventServiceTest.java [hunk 14]
```diff
@@ -247,4 +263,98 @@ void getForUser() {
...
+
+    // then
+    assertTrue(result);
+    assertEquals(EventStatus.CANCELLED, event.getStatus());
+  }
+
+  @Test
+  public void deleteEvent_shouldReturnFalse_whenInvlalidId() {
+    // given
+    final String ORGANIZER_EMAIL = "organizer@example.com";
```
Comment (adamdab) [inline src/test/java/com/team3/central/services/EventServiceTest.java]: We need to make naming convention consistent, change this naming style here or for all tests please

File: src/main/java/com/team3/central/controllers/EventsApiImpl.java (hunk 7 above)
Comment (CJ4n) [reply src/main/java/com/team3/central/controllers/EventsApiImpl.java]: all of the above: applying code style does this

Comment (adamdab) [reply src/main/java/com/team3/central/controllers/EventsApiImpl.java]: Revert please, let's keep git history as clean as possible

Comment (CJ4n) [reply src/main/java/com/team3/central/controllers/EventsApiImpl.java]: ok

**Review by adamdab — APPROVED** [review]

**Pull request merged**